src/accelerate/utils/modeling.py

```diff
@@ -377,13 +379,40 @@ def named_module_tensors(module: nn.Module, include_buffers: bool = True, recurs
             Whether or not to include the buffers in the result.
         recurse (`bool`, *optional`, defaults to `False`):
             Whether or not to go look in every submodule or just return the direct parameters and buffers.
+        remove_non_persistent (`bool`, *optional*, defaults to `False`):
+            Whether or not to remove the non persistent buffer from the buffers. Useful only when include_buffers =
+            True
     """
     for named_parameter in module.named_parameters(recurse=recurse):
         yield named_parameter
 
     if include_buffers:
+        non_persistent_buffers = set()
+        if remove_non_persistent:
+            non_persistent_buffers = get_non_persistent_buffers(module, recurse=recurse)
         for named_buffer in module.named_buffers(recurse=recurse):
-            yield named_buffer
+            name, _ = named_buffer
+            if name not in non_persistent_buffers:
+                yield named_buffer
+
+
+def get_non_persistent_buffers(module: nn.Module, recurse: bool = False):
+    """
+    Gather all non persistent buffers of a given modules into a set
+
+    Args:
+        module (`nn.Module`):
+            The module we want the non persistent buffers on.
+        recurse (`bool`, *optional*, defaults to `False`):
+            Whether or not to go look in every submodule or just return the direct non persistent buffers.
+    """
+
+    non_persistent_buffers_set = module._non_persistent_buffers_set
```

Comment:
We have an issue here. This capability was only added in torch 1.13, and accelerate requires torch 1.10/the last 2 years of torch releases. We should do a guard here for the time being + modify the logic as such. https://github.com/pytorch/pytorch/commit/817eb94ff494e07e94a9effb626ce2121f226a54

Reply:
Good catch ! Thanks for looking into =) I will add a guard.

Reply:
Looks like in this commit, the arg was already present. From the torch 1.10 `[module.py](https://github.com/pytorch/pytorch/blob/v1.10.0/torch/nn/modules/module.py)` file, the arg is also there. So I think we are good. LMK if there is something I missed. Thanks for looking into it, I forgot to do it beforehand!

Reply:
Good catch! LG2M then!